### DISCUSSION TITLE
UI not working with url

### DISCUSSION THREAD

**Discussion by joszon** [gerbera/gerbera#2062] [Q&A] [answered]
I use V1.9.2-77 and it is not possible to reach Gerbera using an URL. 
The UI works with the IP address (192.168.23.21).
(I don't want to expose the URL so that I have replaced the URL that I use with Bogus Media.domain.com) 
config.xml:
```
<webroot>/usr/local/share/gerbera/web</webroot>
<virtualURL>http://media.domain.com:49152</virtualURL>
<port>49152</port>
```
curl -v  media.domain.com:49152
```
* Rebuilt URL to: media.domain.com:49152/
*   Trying 192.168.23.21...
* TCP_NODELAY set
* Connected to media.domain.com (192.168.23.21) port 49152 (#0)
> GET / HTTP/1.1
> Host: media.domain.com:49152
> User-Agent: curl/7.58.0
> Accept: */*
>
* Empty reply from server
* Connection #0 to host media.domain.com left intact
```
what is going on?

**Comment by whyman** [comment]
There was a security fix in libupnp/npupnp recently which forces only IP names to avoid a possible DNS rebind issue.

**Reply by joszon** [reply]
Is it possible that this works again in the near future?
Do I need extra configuration on my local DNS or server to get this working?
Why does Gebera generates no error message that IP is enforced?

**Reply by whyman** [reply]
This check is enforced at the libupnp level, so your requests doesnt make it to the Gerbera code for us to do anything about it.

I think it is possible to allow hostnames with newer versions of the library.

I will open an issue for that, but as @KarlStraussberger mentions, a reverse proxy is the best idea as you can also then do TLS for the UI.

**Comment by KarlStraussberger** [comment, marked as answer]
The best solution is to hide it behind a reverse proxy. This way you can even provide a certificate:
https://docs.gerbera.io/en/stable/run.html#reverse-proxy-setup
